File: Java/PrimaryArithmetic.java
import java.util.*;

public class PrimaryArithmetic {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int a = in.nextInt(), b = in.nextInt();
		int carry, count;
		while(a>0||b>0) {
			carry = 0;
			count = 0;
			while(a>0||b>0) {
				if (a%10 + b%10 + carry > 9) {
					count++;
					carry = 1;
				} else carry = 0;
				a/=10;
				b/=10;
			}
			if(count == 0)
				System.out.println("No carry operation.");
			else if(count == 1)
				System.out.println("1 carry operation.");
			else
				System.out.println(count + " carry operations.");
			a = in.nextInt();
			b = in.nextInt();
		}	
	}
}

File: Cpp/primaryarithmetic.cpp
#include <iostream>
using namespace std;

void fastint(int &num) {
	int c;
	num = 0;
	c = getchar();
	for(;c>47&&c<58;c=getchar()) num=num*10+c-48;
}

int main() {
	int a,b;
	fastint(a);
	fastint(b);
	int count,lcarry;
	while(a||b) {
		count = 0;
		lcarry = 0;
		while(a||b) {
			if (a%10 + b%10 + lcarry > 9) {
				count++;
				lcarry=1;		
			} else lcarry = 0;
			a/=10;
			b/=10;
		}	
		if(count == 1) {
			cout<<count<<" carry operation.\n";
		} else if(count == 0) {
			 cout<<"No carry operation.\n";
		} else cout<<count<<" carry operations.\n";
		fastint(a);
		fastint(b);
	}
	return 0;
}
